types/vue-meta.d.ts

```diff
@@ -0,0 +1,118 @@
+import './vue'
+import Vue, { ComponentOptions, PluginFunction, PluginObject } from 'vue'
+
+type Component = ComponentOptions<Vue> | typeof Vue
+type elements = HTMLElement[]
+
+export interface VueMetaOptions {
+  keyName: string, // the component option name that vue-meta looks for meta info on.
+  attribute: string, // the attribute name vue-meta adds to the tags it observes
+  ssrAttribute: string, // the attribute name that lets vue-meta know that meta info has already been server-rendered
+  tagIDKeyName: string // the property name that vue-meta uses to determine whether to overwrite or append a tag
+  refreshOnceOnNavigation: boolean
+}
+
+export declare class VueMeta {
+  static version: string
+  static install(vue: typeof Vue, options?: VueMetaOptions): PluginFunction<never>
+  static hasMetaInfo(vm: Component): boolean
+}
+
+interface Refreshed {
+  vm: Component,
+  metaInfo: MetaInfo,
+  tags: {
+    addedTags: elements
+    removedTags: elements
+  }
+}
+
+export interface VueMetaPlugin {
+  getOptions(): VueMetaOptions
+  refresh(): Refreshed
+  inject(): MetaInfoSSR
+  pause(refresh: true): () => Refreshed
+  pause(refresh?: boolean): () => void
+  resume(refresh: true): Refreshed
+  resume(refresh?: boolean): void
+}
+
+export interface MetaInfo {
+  title?: string
+  titleTemplate?: string | ((titleChunk: string) => string)
+  htmlAttrs?: { [key: string]: string | string[] }
+  headAttrs?: { [key: string]: string | string[] }
+  bodyAttrs?: { [key: string]: string | string[] }
+  base?: { target: string, href: string }
+
+  meta?: {
```

Comment:
Not mandatory but might be better to have more interfaces and have things like 

```ts
meta?: FindAName[]
link?: FindAName[]
...
```
to prevent `{ ... }[]`

Reply:
Will have a look at that, but your `FindAName` is exactly the issue here and now I didnt have too :)

(and also it was already like this I believe)

Reply:
It's also fine to keep it as it is now, don't take to much time on this :)

Reply:
The only reason I might have a look at this is because we will probably need these types anyway as interfaces for VueMeta 3.0. So we could do it now already.

I was actually searching if someone had already done this, these interfaces describe the properties of standard HTMLElements, surely somebody was in the same situation before and already implemented this. Was hoping I could ~~steal~~ ~~copy~~ borrow their work ;)